code/Control/UserDefinedFormController.php

```diff
@@ -511,6 +534,231 @@ public function finished()
         ]);
     }
 
+    /**
+     * Returns a TextField for entering a folder name.
+     * @param string $folder The current folder to set the field to
+     * @param string $title The title of the text field
+     * @return TextField
+     */
+    private static function getRestrictedAccessField(string $folder, string $title) {
+        /** @var TextField $textField */
+        $textField = TextField::create('CreateFolder', '');
+
+        /** @var Folder $formSubmissionsFolder */
+        $formSubmissionsFolder = Folder::find($folder);
+        $textField->setDescription(EditableFileField::getFolderPermissionString($formSubmissionsFolder));
+        $textField->addExtraClass('pt-2');
+        $textField->setSchemaData([
+            'data' => [
+                'prefix' => static::config()->get('form_submissions_folder') . '/',
+            ],
+            'attributes' => [
+                'placeholder' => $title
+            ]
+        ]);
+
+        return $textField;
+    }
+
+    /**
+     * This returns a Confirm Folder form used to verify the upload folder for EditableFileFields
+     * @return ViewableData
+     */
+    public function confirmfolderform()
+    {
+        $request = $this->getRequest();
+        $id = $request->requestVar('ID');
+        if (!$id) {
+            throw new HTTPResponse_Exception(_t(__CLASS__.'.INVALID_REQUEST', 'This request was invalid.'), 400);
+        }
+        $userFormID = $request->requestVar('UserFormID');
+        if (!$userFormID) {
+            throw new HTTPResponse_Exception(_t(__CLASS__.'.INVALID_REQUEST', 'This request was invalid.'), 400);
+        }
+        $userForm = UserDefinedForm::get()->byID($userFormID);
+        if (!$userForm) {
+            $userForm = Versioned::get_by_stage(UserDefinedForm::class, Versioned::DRAFT)->byID($userFormID);
+        }
+        if (!$userForm) {
+            throw new HTTPResponse_Exception(_t(__CLASS__.'.INVALID_REQUEST', 'This request was invalid.'), 400);
+        }
+
+        if (!$userForm->canView()) {
```

Comment:
Can't think of a reason why you should be able to access this action if you don't have permission to edit the page.

```suggestion
        if (!$userForm->canEdit()) {
```